Fix for the name change bug

NoahRosa opened the pull request:
This PR should fix #6 with two limitations:

1. The device name can't be more than 7 characters long. If the string passed to the begin function is longer, it is automatically trimmed to 7 characters.
2. The android application designed to work with the SensorDemo example expects the name BlueNRG. Otherwise the device can't be opened.

cparata commented:
@fpistm , what about this PR? Could we integrate it in the master repository?

fpistm commented on src/sensor_service.h:
```diff
@@ -149,6 +149,7 @@ class SensorServiceClass
     volatile uint32_t    press_data;
     volatile uint16_t   hum_data;
 
+	char dev_name[8];
```
Please, convert tab to space for alignment

fpistm commented on examples/SPBTLE_SensorDemo/SPBTLE_SensorDemo.ino:
```diff
@@ -39,7 +39,7 @@ SPIClass BTLE_SPI(PIN_BLE_SPI_MOSI, PIN_BLE_SPI_MISO, PIN_BLE_SPI_SCK);
 SPBTLERFClass BTLE(&BTLE_SPI, PIN_BLE_SPI_nCS, PIN_BLE_SPI_IRQ, PIN_BLE_SPI_RESET, PIN_BLE_LED);
 
-const char *name = "BlueNRG";
+const char *name = "BlueNRG"; \\Should be at most 7 characters
```
Is it compile? As I see `\\` instead of `//` for comment.

fpistm commented:
Additionnal comment, maybe the restriction could be added in the README.md and/or sketch header?

NoahRosa commented:
I modified the commit in order to include the changes you requested.
I added the limitation description both in the README.md and the sketch header.
Sorry about the comment, I added it as the last thing and forgot to recompile.

fpistm commented:
No worries Noah. This also happen to me. 😉

NoahRosa committed:
Fix for the name change bug

fpistm commented on src/sensor_service.cpp:
```diff
@@ -126,6 +126,11 @@ tBleStatus SensorServiceClass::begin(const char *name, uint8_t addr[BDADDR_SIZE]
+  memset(dev_name, 0, sizeof(dev_name));
+  
+  dev_name[0] =AD_TYPE_COMPLETE_LOCAL_NAME;
+  strncpy(&dev_name[1], name, (strlen(name)<7) ? strlen(name) : 7);
```
Hereafter a check is done on `(name == NULL)`
I  think this new part of code should be after this sketch.
To go further, maybe if the name is NULL, we could use the default `dev_name`?

```
  uint8_t dev_nameLen = 7; // default 
  if(addr == NULL) {
    return BLE_STATUS_NULL_PARAM;
  }
  if(name != NULL) {
    memset(dev_name, 0, sizeof(dev_name)); 
    dev_nameLen = (strlen(name)<7) ? strlen(name) : 7;
    dev_name[0] =AD_TYPE_COMPLETE_LOCAL_NAME;
    strncpy(&dev_name[1], name, dev_nameLen );
  }
```

Note that I've added `dev_nameLen` to use it later.
Anyway, look my comment about which parameter should be effectively used: https://github.com/stm32duino/SPBTLE-RF/pull/7#discussion_r272454381

NoahRosa replied:
I modified the code in order to work after the not null condition: as you suggested I defined a constant to use when the name is not defined.

fpistm commented on src/sensor_service.cpp:
```diff
@@ -190,7 +195,7 @@ tBleStatus SensorServiceClass::begin(const char *name, uint8_t addr[BDADDR_SIZE]
 
   ret = aci_gatt_update_char_value(service_handle, dev_name_char_handle, 0,
-                                   strlen(name), (uint8_t *)name);
+                                   (strlen(name)<7) ? strlen(name) : 7, (uint8_t *)name);
```
If the NULL is handle like suggested in my previous comment,
then here use the `dev_name`
```suggestion
                                   dev_nameLen , (uint8_t *)&dev_name[1]);
```

NoahRosa replied:
I integrated the change you suggested

fpistm commented on src/sensor_service.cpp:
```diff
@@ -523,15 +528,13 @@ void SensorServiceClass::setConnectable(void)
 
     ret = aci_gap_set_discoverable(ADV_IND, 0, 0, PUBLIC_ADDR, NO_WHITE_LIST_USE,
-                                   sizeof(local_name), local_name, 0, NULL, 0, 0);
+                                   strlen(dev_name), dev_name, 0, NULL, 0, 0);
```
I think there is an issue here and/or maybe in the `begin()`
`dev_name` is not a NULL terminated string so the `strlen()` will return a wrong value, right?
Moreover, previously the size of the `local_name` was used (8) while in the begin() the `dev_name[0]` is not considered ? I know function are not the same, but I think this is something to verify because looking those function in aci gatt/gap, it seems handled to the same way.

Edit: looking the description header of each functions, it seems OK. Anyway use of `strlen` still seems not correct and the question is now which length to uses 8 or the AD + real name length?

NoahRosa replied:
You are definitely right about strlen, I substituted it with AD + real name because it seemed to me to be more correct.
About the difference between the two functions I agree with your perplexities, but I am not experienced enough in the bluetooth stack to give you a certain feedback. I followed the function structure as it was written before but I can't be sure it's the correct/better approach.

fpistm replied:
Thanks @NoahRosa.
@cparata any thought on this?

cparata replied:
For me it is fine. Could you merge and make a new release for the library?

fpistm replied:
ok, you can merge and release.

Commit:
Add requested changes

Commit:
Incremented version
